petastorm/pytorch.py

```diff
@@ -131,15 +131,9 @@ def __init__(self, reader, batch_size=1, collate_fn=decimal_friendly_collate,
 
         # _batch_acc accumulates samples for a single batch.
         self._batch_acc = []
-        if shuffling_queue_capacity > 0:
-            # We can not know what is the reasonable number to use for the extra capacity, so we set a huge number
-            # and give up on the unbound growth protection mechanism.
-            min_after_dequeue = shuffling_queue_capacity - 1
-            self._shuffling_buffer = RandomShufflingBuffer(shuffling_queue_capacity,
-                                                           min_after_retrieve=min_after_dequeue,
-                                                           extra_capacity=100000000)
-        else:
-            self._shuffling_buffer = NoopShufflingBuffer()
+        self.shuffling_queue_capacity = shuffling_queue_capacity
+        self._in_iter = False
+        self._iter_pass = 0
```

Comment:
Do we need to remember the number of passes? We can set `_in_iter` to (None, False, True) to tell whether it is the first pass.